numpy/testing/_private/utils.py

```diff
@@ -876,6 +880,9 @@ def assert_array_equal(x, y, err_msg='', verbose=True):
         The error message to be printed in case of failure.
     verbose : bool, optional
         If True, the conflicting values are appended to the error message.
+    strict : bool, optional
+        If True, raise an assertion when one of the array_like objects is a
+        scalar.
```

Comment:
Is this only true for scalars, or does it disable the more general broadcast rules?

Reply:
Yes, currently only scalars are accepted as equal, so I think they are disabled already. `1 vs np.array([1, 1])` is equal, but `[1] vs np.array([1, 1])` is not.